### PR TITLE
Add more output formats

### LINKED ISSUE #106: Dump json to stdout
It would be helpful to dump json function spaces to `stdout`

### PR DESCRIPTION
**Pull request by Luni-4**
This PR closes #106 and fixes some items of #104 

- Add `cbor`
- Add `json`
- Add `toml`
- Add `yaml`

Thanks in advance for your review! :)

### CODE REVIEW

**Comment by calixteman** [comment]
Please move all the dump/write stuff in an other file.
Probably metrics.rs should move in the metrics directory.

**Comment by Luni-4** [comment]
@calixteman 

This PR is done now!

File: rust-code-analysis-cli/src/main.rs [hunk 1]
```diff
@@ -303,11 +318,19 @@ fn main() {
                 .takes_value(true),
         )
         .arg(
-            Arg::with_name("type")
+            Arg::with_name("language_type")
                 .help("Language type")
+                .short("l")
+                .long("lt")
```
Comment (calixteman) [inline rust-code-analysis-cli/src/main.rs:324]: nit: --language-type for the long version is more explicit

File: rust-code-analysis-cli/src/main.rs [hunk 2]
```diff
@@ -303,11 +318,19 @@ fn main() {
...
+                .default_value("")
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("output_format")
+                .help("Output metrics as different formats")
                 .short("t")
                 .long("type")
```
Comment (calixteman) [inline rust-code-analysis-cli/src/main.rs:332]: nit:--output-format for the long and -O for the short.

File: src/enums.rs [hunk 3]
```diff
@@ -0,0 +1,30 @@
+use std::fmt;
+
+#[derive(Clone, Copy, Debug, PartialEq, Serialize)]
+#[serde(rename_all = "lowercase")]
+pub enum NodeKind {
```
Comment (calixteman) [inline src/enums.rs:5]: you probably need to rebase

File: src/output/dump_metrics.rs [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+use crate::exit;
+use crate::fn_args;
+use crate::halstead;
+use crate::loc;
+use crate::mi;
+use crate::nom;
+
+use crate::spaces::{CodeMetrics, FuncSpace};
+
+pub(crate) fn dump_root(space: &FuncSpace) -> std::io::Result<()> {
```
Comment (calixteman) [inline src/output/dump_metrics.rs:14]: is it still possible to dump using this stuff ?

Comment (Luni-4) [reply src/output/dump_metrics.rs:14]: I can't understand the question. Could you elaborate more on this one please?

Comment (calixteman) [reply src/output/dump_metrics.rs:14]: I tried the patch locally and it's ok.
My question was just to know if it's still possible to dump metrics on stdout in using the code here. 
And using single --metrics is ok.

**Comment by calixteman** [comment]
I made a stupid test: `cargo run -- -p ../../mozilla-central.hg/dom/animation/Animation.cpp --metrics -t json | jq` and it failed:
`parse error: Invalid numeric literal at line 1, column 2`
Could you fix it please ?

File: src/output/dump_metrics.rs (hunk 4 above)
Comment (Luni-4) [reply src/output/dump_metrics.rs:14]: Oh, I see. I'm glad you solved it :)

File: rust-code-analysis-cli/src/main.rs [hunk 5]
```diff
@@ -303,13 +318,21 @@ fn main() {
...
+                .short("l")
+                .long("language-type")
                 .default_value("")
                 .takes_value(true),
         )
+        .arg(
+            Arg::with_name("output_format")
+                .help("Output metrics as different formats")
+                .short("O")
+                .long("--output-format")
```
Comment (calixteman) [inline rust-code-analysis-cli/src/main.rs:332]: nit: '--' are useless

**Comment by Luni-4** [comment]
@calixteman 

Can this one be merged?

**Pull request merged**